src/libraries/System.Net.Http/src/System/Net/Http/HttpMethod.cs

```diff
@@ -175,13 +171,29 @@ public override string ToString()
         /// </summary>
         internal static HttpMethod Normalize(HttpMethod method)
         {
-            // _http3EncodedBytes is only set for the singleton instances, so if it's not null,
+            // _http3Index is only set for the singleton instances, so if it's not null,
             // we can avoid the dictionary lookup.  Otherwise, look up the method instance in the
             // dictionary and return the normalized instance if it's found.
-            Debug.Assert(method != null);
-            return
-                method._http3EncodedBytes is null && s_knownMethods.TryGetValue(method, out HttpMethod? normalized) ? normalized :
-                method;
+            if (method._http3Index != null)
+                return method;
+
+            if (s_knownMethods is null)
+            {
+                s_knownMethods = new Dictionary<HttpMethod, HttpMethod>(9)
```

Comment:
Technically this should use a Volatile.Read/Write:
```C#
Dictionary<HttpMethod, HttpMethod>? knownMethods = Volatile.Read(ref s_knownMethods);
if (knownMethods is null)
{
    knownMethods = new Dictionary<...>(...) { ... };
    Volatile.Write(ref s_knownMethods, knownMethods);
}

return knownMethods.TryGetValue(...
```

Reply:
Given how small this dictionary is, I wonder if we shouldn't just write some hard-wired code to check for the specific methods. That would mean less memory usage (no dictionary) and would probably be faster as well.

Reply:
Yeah, we should:
https://github.com/dotnet/runtime/pull/47281